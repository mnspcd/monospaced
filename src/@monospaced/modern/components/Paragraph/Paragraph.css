.mdrn .Paragraph {
  margin-bottom: 0;
  margin-top: 0;
  max-width: var(--measure);
}
