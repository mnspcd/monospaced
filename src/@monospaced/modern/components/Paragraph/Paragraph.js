import PropTypes from "prop-types";
import React from "react";

import "./Paragraph.css";

/**
 * Use `Paragraph` to typeset text in paragraphs.
 */
const Paragraph = ({ children }) => <p className="Paragraph">{children}</p>;

Paragraph.propTypes = {
  /**
   * Text to display in the Paragraph
   */
  children: PropTypes.node.isRequired,
};

export default Paragraph;
