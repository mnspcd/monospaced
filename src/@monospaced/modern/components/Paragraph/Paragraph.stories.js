import { text } from "@storybook/addon-knobs";
import React from "react";

import Paragraph from "./Paragraph";

export default {
  component: Paragraph,
  title: "Components/Paragraph",
};

export const basic = () => (
  <Paragraph>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat.
  </Paragraph>
);

export const knobs = () => (
  <Paragraph>
    {text(
      "text",
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
    )}
  </Paragraph>
);

knobs.story = { parameters: { docs: { disable: true } } };
