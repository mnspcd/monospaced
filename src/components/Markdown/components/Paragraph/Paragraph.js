import PropTypes from "prop-types";
import React from "react";

import "./Paragraph.css";

const Paragraph = ({ children }) => <p className="Paragraph">{children}</p>;

Paragraph.propTypes = { children: PropTypes.node.isRequired };

export default Paragraph;
