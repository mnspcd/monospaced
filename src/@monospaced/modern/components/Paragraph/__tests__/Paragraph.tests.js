/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Paragraph from "..";

describe("Paragraph component", () => {
  it("should render correctly", () => {
    const { container } = render(<Paragraph>children</Paragraph>);

    expect(container).toMatchSnapshot();
  });
});
