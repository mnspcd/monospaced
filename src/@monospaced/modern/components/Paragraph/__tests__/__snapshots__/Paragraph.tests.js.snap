// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph component should render correctly 1`] = `
<div>
  <p
    class="Paragraph"
  >
    children
  </p>
</div>
`;
